MoviesforSale/app/src/main/java/com/ezequielc/moviesforsale/MovieListFragment.java

```diff
@@ -0,0 +1,127 @@
+package com.ezequielc.moviesforsale;
+
+import android.app.SearchManager;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v4.view.MenuItemCompat;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.List;
+
+
+public class MovieListFragment extends Fragment {
+    private OnMovieSelectedListener mListener;
+    private MovieAdapter mMovieAdapter;
+    private List<Movies> mMoviesToDisplay;
+
+
+    public MovieListFragment() {
+        // Required empty public constructor
+    }
+
+    public  interface OnMovieSelectedListener{
+        void onMovieSelected(int selectedMovie);
+    }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.options_menu, menu);
+
+        SearchManager searchManager = (SearchManager) getContext().getSystemService(Context.SEARCH_SERVICE);
+        MenuItem menuItem = menu.findItem(R.id.search);
+        SearchView searchView = (SearchView) menu.findItem(R.id.search).getActionView();
+        ComponentName componentName = new ComponentName(getContext(), MainActivity.class);
+        searchView.setSearchableInfo(searchManager.getSearchableInfo(componentName));
+
+        MenuItemCompat.setOnActionExpandListener(menuItem, new MenuItemCompat.OnActionExpandListener() {
+            @Override
+            public boolean onMenuItemActionExpand(MenuItem item) {
+                return true;
+            }
+
+            @Override
+            public boolean onMenuItemActionCollapse(MenuItem item) {
+                // Re-populates RecyclerView
+                mMoviesToDisplay.clear();
```

Comment:
When I perform a search, then close the `SearchView` and this method runs, I get a crash on this line with a null pointer exception, because `mMoviesToDisplay` was never initialized. This is because in `onViewCreated()` you declared a new, local variable to hold your list of movies rather than use `mMoviesToDisplay`